client/fingerprint/env_aws.go

```diff
@@ -191,15 +194,20 @@ func (f *EnvAWSFingerprint) linkSpeed(ec2meta *ec2metadata.EC2Metadata) int {
 	return netSpeed
 }
 
-func ec2MetaClient(endpoint string, timeout time.Duration) *ec2metadata.EC2Metadata {
+func ec2MetaClient(endpoint string, timeout time.Duration) (*ec2metadata.EC2Metadata, error) {
 	client := &http.Client{
 		Timeout:   timeout,
 		Transport: cleanhttp.DefaultTransport(),
 	}
 
-	c := aws.NewConfig().WithHTTPClient(client)
+	c := aws.NewConfig().WithHTTPClient(client).WithMaxRetries(0)
 	if endpoint != "" {
 		c = c.WithEndpoint(endpoint)
 	}
-	return ec2metadata.New(session.New(), c)
+
+	session, err := session.NewSession(c)
+	if err != nil {
+		return nil, err
+	}
+	return ec2metadata.New(session, c), nil
```

Comment:
Curious, if the session is already configured with the config option, do we still need to pass it to `ec2metadata.New`?

Reply:
I am honestly not quite sure, it's probably safer to pass to both, because the SDK is a little bit messy tbh.